ui/test/nuxeo-layout.test.js

```diff
@@ -27,57 +35,127 @@ const base = url.substring(0, url.lastIndexOf('/'));
+      expect(nuxeoError.code).to.equal('404');
+      expect(nuxeoError.url).to.equal('notfound.html');
+      expect(nuxeoError.message).to.equal('Failed to find layout');
+    });
+
+    test('Should hide layout when href is empty', async () => {
+      const layout = await fixture(
+        html`
+          <nuxeo-layout href=""></nuxeo-layout>
+        `,
+      );
+      await flush();
 
-  test('layout valid field valid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "valid", "required": "foo"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(true);
+      expect(isElementVisible(layout)).to.be.false;
+    });
   });
 
-  test('layout invalid field valid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid", "required": "foo"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+  suite('Stamping', () => {
+    test('Should stamp the layout when model is defined', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "dummy"}'></nuxeo-layout>
+      `);
+      await flush();
+
+      if (!layout.element) {
+        await waitChanged(layout, 'element');
+      }
+
+      expect(isElementVisible(layout.$.error)).to.be.false;
+      expect(layout.element.localName).to.equal('dummy-layout');
+      expect(layout.element.$$('span')).to.exist;
+      expect(layout.element.$$('span').innerText).to.equal('dummy');
+      expect(layout.element.$$('nuxeo-input')).to.exist;
+    });
+
+    test('Should stamp the layout when model is empty', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/dummy-layout.html" model=""></nuxeo-layout>
+      `);
+      await flush();
+
+      if (!layout.element) {
+        await waitChanged(layout, 'element');
+      }
+
+      expect(layout.element.$$('span')).to.exist;
+      expect(layout.element.$$('nuxeo-input')).to.exist;
+    });
   });
 
-  test('layout invalid field invalid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+  suite('Validation', () => {
+    test('Should validate the layout when fields are valid', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "valid", "data": "foo"}'></nuxeo-layout>
+      `);
+      await flush();
+
+      if (!layout.element) {
+        await waitChanged(layout, 'element');
+      }
+      const validity = await layout.validate();
+      sinon.spy(layout.element, 'validate');
```

Comment:
I understand the reason why we are spying the method, but we need to check if it was called in fact. Probably just one time in this case, correct? 🙂